bolt/include/bolt/Profile/YAMLProfileReader.h

```diff
@@ -105,7 +105,7 @@ class YAMLProfileReader : public ProfileReaderBase {
   yaml::bolt::BinaryProfile YamlBP;
 
   /// Map a function ID from a YAML profile to a BinaryFunction object.
-  std::vector<BinaryFunction *> YamlProfileToFunction;
+  DenseMap<uint32_t, BinaryFunction *> YamlProfileToFunction;
```

Comment:
I'm curious if using `uint32_t` leads to more compact allocation compared to `uint64_t`?

Reply:
I don't think it does: DenseMap stores objects in buckets where bucket is a KV pair where the second element (pointer) would force the alignment to be 8b.

Reply:
But the YamlBF.Id is uint32, so I decided to match it: https://github.com/llvm/llvm-project/blob/6b21cf8ccad84e2670e458d8bdaccbd0ae37b46b/bolt/include/bolt/Profile/ProfileYAMLMapping.h#L200-L203